src/helix-ui/elements/HXTooltipElement.js

```diff
@@ -38,173 +27,284 @@ export class HXTooltipElement extends HXElement {
         return `<style>${shadowStyles}</style>${shadowMarkup}`;
     }
 
+    /** @override */
     $onCreate () {
-        this._onShow = this._onShow.bind(this);
-        this._onHide = this._onHide.bind(this);
-        this._onClick = this._onClick.bind(this);
-        this._setPosition = this._setPosition.bind(this);
-        this._onDocumentClick = this._onDocumentClick.bind(this);
+        super.$onCreate();
+
+        // overrides Positionable default
+        this.DEFAULT_POSITION = 'top';
+
+        this._onCtrlBlur = this._onCtrlBlur.bind(this);
+        this._onCtrlFocus = this._onCtrlFocus.bind(this);
+        this._onCtrlMouseEnter = this._onCtrlMouseEnter.bind(this);
+        this._onCtrlMouseLeave = this._onCtrlMouseLeave.bind(this);
+        this._onEsc = this._onEsc.bind(this);
+
+        // TODO: What if 'id' is blank?
+        this.$defaultAttribute('id', `tip-${this.$generateId()}`);
+
+        this.$defaultAttribute('role', 'tooltip');
+        this.$defaultAttribute('data-offset', 20);
+        this._isHovering = false;
     }
 
+    /** @override */
     $onConnect () {
-        this.$defaultAttribute('position', 'top');
-        this.initialPosition = this.position;
-        this.$upgradeProperty('open');
-        this.$defaultAttribute('role', 'tooltip');
+        super.$onConnect();
 
-        this.setAttribute('aria-hidden', !this.open);
+        this.$upgradeProperty('for');
 
-        if (this.id) {
-            this._target = this.getRootNode().querySelector(`[data-tooltip="${this.id}"]`);
-        } else {
-            return;
-        }
-        this._connectHandlers();
+        this._attachListeners();
     }
 
+    /** @override */
     $onDisconnect () {
-        if (!this._target) {
-            return;
-        }
-        this._destroyHandlers();
+        super.$onDisconnect();
+
+        this._detachListeners();
     }
 
+    /** @override */
     static get $observedAttributes () {
-        return [ 'open' ];
+        return super.$observedAttributes.concat([ 'for' ]);
     }
 
+    /** @override */
     $onAttributeChange (attr, oldVal, newVal) {
-        if (attr === 'open') {
-            let isOpen = (newVal !== null);
-            this.setAttribute('aria-hidden', !isOpen);
-            this.$emit(isOpen ? 'open' : 'close');
+        super.$onAttributeChange(attr, oldVal, newVal);
+
+        switch (attr) {
+            case 'for':
+                this._attrForChange(oldVal, newVal);
+                break;
+
+            case 'position':
+                this._elRoot.setAttribute('position', newVal);
+                break;
         }
     }
 
     /**
-     * Where to position the menu in relation to its reference element.
+     * External element that controls tooltip visibility.
      *
-     * @default "top"
-     * @type {PositionString}
+     * Returns the first element with an "id" matching the tooltip's "for" value.
+     *
+     * @readonly
```

Comment:
please add an `override`

Reply:
Discovered that `@override` isn't needed.